src/client/datascience/jupyter/notebookStarter.ts

```diff
@@ -154,8 +160,10 @@ export class NotebookStarter implements Disposable {
             } catch (ex) {
                 traceError(`Parsing failed ${connection.baseUrl}`, ex);
             }
+            disposeAllDisposables(disposables);
             return connection;
         } catch (err) {
+            disposeAllDisposables(disposables);
```

Comment:
Why not put this in a finally block?

Reply:
I don't want the catch statements to slow this down, this needs to happen immediately.